@here/harp-materials/lib/ShadowFeature.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { chainCallbacks } from "@here/harp-utils";
+import * as THREE from "three";
+import { HiddenThreeJSMaterialProperties } from "./MapMeshMaterials";
+import { simpleLightingShadowChunk } from "./ShaderChunks/ShadowChunks";
+
+/**
+ * Parameter used to control patching the THREE.MeshStandardMaterial shader to ensure that the
+ * material's color isn't affected by the light direction.
+ */
+export interface ShadowFeatureParameters {
+    /**
+     * Whether the diffuse light component is removed (i.e. the materials color is therefore just
+     * the ambient + shadow).
+     */
+    removeDiffuseLight?: boolean;
+}
+
+/**
+ * Applies changes to THREE's MeshStandardMaterial which removes the diffuse component to ensure
+ * that colors are kept consistent.
+ */
+export namespace ShadowFeature {
+    export function onBeforeCompile(material: HiddenThreeJSMaterialProperties) {
+        material.onBeforeCompile = chainCallbacks(material.onBeforeCompile, shader => {
```

Comment:
Why not saving some bytes ?
This code can be just thrown into MapMeshStandardMaterial contstructor.

Reply:
Ok, will do.